File: app/src/main/java/com/gerosprime/gmoviedb/screens/movies/populars/PopularListView.java
package com.gerosprime.gmoviedb.screens.movies.populars;

import com.gerosprime.gmoviedb.components.mvp.MovieDBMvpView;
import com.gerosprime.gmoviedb.models.movies.populars.PopularMovieEntity;

import java.util.List;

public interface PopularListView
    extends MovieDBMvpView {

    void populatePopulars(List<PopularMovieEntity> movieEntities);
    void hideContentList();
    void showContentList();

    void hideLoadingIndicator();
    void showLoadingIndicator();

    void showContentError();
    void hideContentError();
    void loadingPopularApiError();
    void loadingPopularTechnicalError();

}

File: app/src/main/java/com/gerosprime/gmoviedb/screens/movies/populars/PopularListView.kt
package com.gerosprime.gmoviedb.screens.movies.populars

import com.gerosprime.gmoviedb.components.mvp.MovieDBMvpView
import com.gerosprime.gmoviedb.models.movies.populars.PopularMovieEntity

interface PopularListView : MovieDBMvpView {

    fun populatePopulars(movieEntities: List<PopularMovieEntity>)

    fun hideContentList()
    fun showContentList()

    fun hideLoadingIndicator()
    fun showLoadingIndicator()

    fun showContentError()
    fun hideContentError()

    fun loadingPopularApiError()
    fun loadingPopularTechnicalError()

}